agent/api/task/task_linux.go

```diff
@@ -355,3 +357,38 @@ func (task *Task) BuildCNIConfig(includeIPAMConfig bool, cniConfig *ecscni.Confi
 
 	return cniConfig, nil
 }
+
+// BuildCNIConfigBridgeMode builds a list of CNI network configurations for a task in docker bridge mode.
+func (task *Task) BuildCNIConfigBridgeMode(cniConfig *ecscni.Config, containerName string) (*ecscni.Config, error) {
+	if !task.IsNetworkModeBridge() {
+		return nil, errors.New("non bridge-mode task should not invoke BuildCNIConfigBridgeMode")
+	}
+
+	var netconf *libcni.NetworkConfig
+	var ifName string
+	var err error
+
+	// Build a CNI network configuration for bridge-mode SC-enabled task
+	if task.IsServiceConnectEnabled() {
```

Comment:
If evaluated to false, shouldnt we be retuning error?

Reply:
I was thinking that we may add other bridge-mode CNI plugins in the future so I kept it open. But you are right that we probably should throw error because SC task is the only one that should be invoking this method at this moment.